lib/s3middleware/objectLegalHold.js

```diff
@@ -0,0 +1,111 @@
+const { parseString } = require('xml2js');
+const errors = require('../errors');
+
+/*
+    Format of the xml request:
+    <LegalHold>
+        <Status>ON|OFF</Status>
+    </LegalHold>
+*/
+
+/**
+ * validate status - validates status xml
+ * @param {object} status - parsed status object
+ * @return {boolean} - true or false on success, error on failure
```

Comment:
This doesn't seem right to me, generally speaking, functions should return one type. A method name like validate status to me seems like it would simply throw or return an error but not return anything. Maybe this is more acceptable in javascript than other languages but by returning two different types you force callees to conditional check by type. Where as if you just threw the error it would be a standard try/catch. Or maybe if you returned another object or split this method in two.

Reply:
That makes sense. I tweaked it so that it always returns an object; either with error or with status.